DuckDuckGoTests/XCTestCaseExtension.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  XCTestCaseExtension.swift
+//  DuckDuckGo
+//
+//  Copyright © 2022 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+
+extension XCTestCase {
+    
+    func temporaryUserDefaultSuite(with filePath: String) -> String {
+        guard let lastPathComponent = NSURL(fileURLWithPath: filePath).lastPathComponent else {
+            fatalError("Path should have a last path component")
+        }
+        let temporaryPath = "/var/tmp/"
```

Comment:
Haven't tested this on iOS (Simulator), but there's an API in FileManager for creating temporary directories that are automatically deleted after a process terminates. If you like, see the "Discussion" section [over here](https://developer.apple.com/documentation/foundation/filemanager/1407693-url) for more information.